### PR TITLE
docs: added docs for Mustache template

### LINKED ISSUE #856: Write a guide on mustache templates
Tailcall allows the use of mustache templates in various places of the configuration. Write a guide about the power it gives to the developers and it's limitations.

**Requirements**
- Explain what it does and why do we need it.
- List down all the places where mustache templates can be used.
- Link to `context.md` in the document for further reading.
- Back link to `mustache.md` in the `context.md`.
- Make no grammatical or spelling errors.
- Keep the guide tone neutral, non non-emotional.
- Keep the writing style free-flowing and engaging.
- Fact-check everything, connect on Discord if you are confused.
- Make the writing cohesive, to the point, and related to the title.

**PS: DON'T CREATE LOW-EFFORT CHATGPT GENERATED CONTENT!**

### PR DESCRIPTION
**Pull request by ibishal**
**Summary:**  
Documentation for using mustache template with tailcall

**Issue Reference(s):**  
Fixes #856 
/claim #856

### CODE REVIEW

File: docs/guides/mustache.md [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+```
+
+#### Consistency in Configuration
+
+Consider a case where you're utilizing the same base URL across multiple API endpoints.Mustache templates help us to do this in the simplified form below
+
+```graphql
+type Query {
+  users: [User] @http(path: "/users", baseURL: "{{headers.apiBaseUrl}}")
+  posts: [Post] @http(path: "/posts", baseURL: "{{headers.apiBaseUrl}}")
```
Comment (tusharmath) [inline docs/guides/mustache.md:120]: Using environment variables would be a better example here.

File: docs/guides/mustache.md [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+In the changed `completedTodos` query, we have added a way to ask for either completed or ongoing tasks. When you run this `completedTodos` query in the playground, you can ask for only the completed tasks by typing `completed: true` in the box where you normally put details. This way, you'll get a list that includes only the tasks that are finished.
+
+#### Customization in Queries
+
+Imagine a scenario where you need to fetch paginated data from an API
+
+```graphql
+type Query {
+  paginatedPosts(page: Int!): [Post]
+    @http(path: "/posts", query: [{key: "page", value: "{{args.page}}"}])
```
Comment (tusharmath) [inline docs/guides/mustache.md:71]: You can use mustache in the `path` also.

File: docs/guides/mustache.md [hunk 3]
```diff
@@ -0,0 +1,126 @@
+---
+title: Mustache template with tailcall
+---
+Mustache templates are like fill-in-the-blank in forms for code. They let you put placeholders in your configurations, allowing dynamic content to be inserted. The dynamic content inserted within the curly braces ( {{ }} )
+Learn more about mustache template [here](https://mustache.github.io/)
```
Comment (tusharmath) [inline docs/guides/mustache.md:5]: We should say that where ever you have a string literal used in a directive, it would support a mustache template.

**Comment by ibishal** [comment]
@tusharmath added the changes that you have mentioned

**Comment by tusharmath** [comment]
@ibishal Can you raise this PR wrt our new github repository.

**Comment by ibishal** [comment]
> @ibishal Can you raise this PR wrt our new github repository - https://github.com/tailcallhq/tailcallhq.github.io

done @tusharmath

**Pull request closed without merging**